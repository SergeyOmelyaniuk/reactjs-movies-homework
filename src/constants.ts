import { Language } from './types';

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const translate: any = {
	en: {
		title: 'title',
		placeholder: 'Movies, person, movie theaters',
		categories: [
			{ id: 1, value: 'popular', title: 'Popular' },
			{ id: 2, value: 'top_rated', title: 'Top rated' },
			{ id: 3, value: 'upcoming', title: 'Upcoming' },
		],
	},
	ru: {
		title: 'заголовок',
		placeholder: 'Фильмы, люди, кинотеатры',
		categories: [
			{ id: 1, value: 'popular', title: 'Популярные' },
			{ id: 2, value: 'top_rated', title: 'Топ рейтинга' },
			{ id: 3, value: 'upcoming', title: 'Ближайшие' },
		],
	},
};
