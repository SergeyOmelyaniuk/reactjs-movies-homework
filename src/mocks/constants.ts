import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
