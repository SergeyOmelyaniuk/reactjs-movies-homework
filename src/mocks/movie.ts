import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
