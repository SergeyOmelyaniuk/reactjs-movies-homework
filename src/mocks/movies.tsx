import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
