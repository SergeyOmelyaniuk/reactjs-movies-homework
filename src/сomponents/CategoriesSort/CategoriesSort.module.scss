.categoriesSort {
	display: inline-flex;
	overflow: hidden;
	border-radius: 10px;
	border: 1px solid rgb(255, 255, 255);
	background-color: #151515;
	& input {
		display: none;
		&:checked + label {
			background-color: #fff;
			color: #000;
		}
	}
	& label {
		font-size: 16px;
		padding: 10px 20px;
		color: #fff;
		cursor: pointer;
		&:not(:last-of-type) {
			border-right: 1px solid rgb(255, 255, 255);
		}
	}
}
