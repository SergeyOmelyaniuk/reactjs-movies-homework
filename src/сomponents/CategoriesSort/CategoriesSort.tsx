import React, { Fragment, MouseEvent } from 'react';
import { Category } from '../../types';
import styles from './CategoriesSort.module.scss';

interface CategoriesSortProps {
	categories: Category[];
	categoryValue: string;
	changeCategory: (value: string) => void;
}

const CategoriesSort = ({
	categories,
	categoryValue,
	changeCategory,
}: CategoriesSortProps) => {
	const handler = (event: MouseEvent) => {
		changeCategory((event.target as HTMLInputElement).value);
	};

	return (
		<div className={styles.categoriesSort}>
			{categories.map((category) => {
				return (
					<Fragment key={category.id}>
						<input
							onClick={handler}
							id={`radio-${category.id}`}
							type='radio'
							name='sort'
							value={category.value}
							defaultChecked={category.value === categoryValue}
						/>
						<label htmlFor={`radio-${category.id}`}>{category.title}</label>
					</Fragment>
				);
			})}
		</div>
	);
};

export default CategoriesSort;
