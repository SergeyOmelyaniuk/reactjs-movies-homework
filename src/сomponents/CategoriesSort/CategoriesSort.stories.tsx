import React from 'react';
import { action } from '@storybook/addon-actions';

import { ComponentStory, ComponentMeta } from '@storybook/react';
import CategoriesSort from './CategoriesSort';
import { categories } from '../../mocks/constants';

export default {
	title: 'Components/CategoriesSort',
	component: CategoriesSort,
	argTypes: {},
} as ComponentMeta<typeof CategoriesSort>;

const Template: ComponentStory<typeof CategoriesSort> = (args) => (
	<div
		style={{
			width: '410px',
			height: '110px',
			backgroundColor: '#202020',
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
		}}
	>
		<CategoriesSort {...args} />
	</div>
);

export const Default = Template.bind({});
Default.args = {
	categories: categories,
	categoryValue: categories[0].value,
	changeCategory: action('Change category'),
};
