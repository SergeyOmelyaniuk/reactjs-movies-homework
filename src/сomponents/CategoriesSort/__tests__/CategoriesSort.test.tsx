import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CategoriesSort from '..';
import { categories } from '../../../mocks/constants';

describe('CategoriesSort', () => {
	test('should render', () => {
		const { container } = render(
			<CategoriesSort
				categories={categories}
				categoryValue={categories[0].value}
				changeCategory={() => {}}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	test('should call change category on click event', () => {
		const changeCategoryMock = jest.fn();
		const { getByLabelText } = render(
			<CategoriesSort
				categories={categories}
				categoryValue={categories[0].value}
				changeCategory={changeCategoryMock}
			/>
		);

		const secondCategoryElement = getByLabelText(categories[1].title);
		fireEvent.click(secondCategoryElement);

		expect(changeCategoryMock).toHaveBeenCalledTimes(1);
		expect(changeCategoryMock).toHaveBeenCalledWith(categories[1].value);
	});
});
