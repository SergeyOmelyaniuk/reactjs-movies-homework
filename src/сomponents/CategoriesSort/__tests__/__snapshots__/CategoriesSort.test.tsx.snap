// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesSort should render 1`] = `
<div>
  <div
    class="categoriesSort"
  >
    <input
      checked=""
      id="radio-1"
      name="sort"
      type="radio"
      value="popular"
    />
    <label
      for="radio-1"
    >
      Popular
    </label>
    <input
      id="radio-2"
      name="sort"
      type="radio"
      value="top_rated"
    />
    <label
      for="radio-2"
    >
      Top rated
    </label>
    <input
      id="radio-3"
      name="sort"
      type="radio"
      value="upcoming"
    />
    <label
      for="radio-3"
    >
      Upcoming
    </label>
  </div>
</div>
`;
